3467/0/triangle.rb

```diff
@@ -0,0 +1,39 @@
+# Triangle Project Code.
+
+# Triangle analyzes the lengths of the sides of a triangle
+# (represented by a, b and c) and returns the type of triangle.
+#
+# It returns:
+#   :equilateral  if all sides are equal
+#   :isosceles    if exactly 2 sides are equal
+#   :scalene      if no sides are equal
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+# :reek:FeatureEnvy
+def triangle(s_a, s_b, s_c)
+  triangle_does_not_exists(s_a, s_b, s_c)
+
+  return :equilateral if (s_a == s_b) && (s_b == s_c)
+  return :isosceles if (s_a == s_b) || (s_b == s_c) || (s_c == s_a)
+
+  :scalene
+end
+
+def triangle_does_not_exists(s_a, s_b, s_c)
+  s_a_b = s_a + s_b
```

Comment:
what about -10 -10 -10?

Reply:
In this case all conditions (-10 + -10 <= -10  =  -20 <= -10)  will be satisfied.